src/HaskellCI/YamlSyntax.hs

```diff
@@ -315,3 +318,35 @@ prettyYaml comment' = flatten . go where
 
 -- a 'Line' is comments before in and actual text after!
 data Line = Line [String] ShowS
+
+escapeYAMLString :: String -> ShowS
```

Comment:
It's not clear to me how you figured out the implementation of this function, which seems crucial to the whole operation here. If there is some reference you used to figure this out, including a link to it would be helpful.

Reply:
I haven't found any. `\u` escaping didn't work.

Reply:
... Long term it would be nice to use `HsYAML` functionality, and not reinvent it.

Reply:
I did find an [`escapeJSON`](https://hackage.haskell.org/package/extra-1.6.19/docs/Data-List-Extra.html#v:escapeJSON) function in the `extra` library, although that appears to be different from `escapeYAMLString` in various ways. For one thing, `escapeJSON` doesn't surround the `String` with double quotes.

If there is no reference, perhaps leave some comments highlighting what things in particular need to be escaped. The `hexChar` function in particular could benefit from this, since I have absolutely no idea what it is doing.

Reply:
Thanks, the new comments help orient myself with what is going on.